### PR TITLE
Minimal notebook UI flag for vscode notebooks

### PR DESCRIPTION
**Pull request by nstrayer**
This PR adds a new mode enabled by the feature flag `notebook.usePositronMinimalNotebookUI`
image

The flag defaults to off. 

## Static changes

There are a few static visual changes:

image

## Animation changes
Perhaps more important are the changes to the experience of navigating and running cells in the notebook.
Previously when you would run a cell the following animations would activate
- Spinner in the  main notebook toolbar
- A _different_ spinner beneath the cell
- The cell execution-time counter
- A bouncing progress bar above the cell. 

Minimal mode removes the cell-level spinner and the bouncing progress bar to keep things a bit less hectic. 

### Running a cell with defaults

https://github.com/user-attachments/assets/82efe13a-d311-4596-82cd-fbd4592f5e29

### Running cells with minimal mode
https://github.com/user-attachments/assets/746d64bb-91ed-46c4-bc96-6569900f49e3

### QA Notes

This PR is almost entirely implemented via css, so there shouldn't be too many hiccups in terms of things changing. Since it's purely UI-based I held off on adding a smoke-test as that felt a bit overboard for this type of change. 
Basically, after changing the feature flag, you should be able to open up any notebook you could open before and have things look a bit more... calm.

### CODE REVIEW

**Comment by dhruvisompura** [comment]
Not sure if the following is a regression from this change but I noticed that the default notebook layout has some issues:
- If your cell only contains a single line of code. It looks like the execute button and `[]` under it are squished together without any margin between the two. Is this a regression or did the default notebook layout always have this issue?
- the buttons to add additional cells (Code and Markdown) don't look to be in consistent places. I have seen the buttons within a cell, straddling between two cells, or under a cell. Is this inconsistency expected?

https://github.com/user-attachments/assets/29dfb9f1-5177-430b-9cd4-8f8041fb7773

https://github.com/user-attachments/assets/029cca5a-fa9f-4df5-9198-2a2b3ac037a7

I did also try to turn the feature flag on but couldn't get the minimal ui to show. It may be user error though. I turned the feature on and then created a new jupyter project and opened it in the current window. Not sure what I did wrong there 😢

**Comment by nstrayer** [comment]
Closing this in favor of focusing work on backend/notebooks 2.0.
